### ISSUE TITLE
Problem with confint.default and glmer: long vectors not supported yet

### ISSUE THREAD

**Issue by bakaburg1** [lme4/lme4#478]
Hello, just recently I am experiencing this problem when using wald confidence intervals with glmer:

> Error in diag(vcov(object)) : 
>   long vectors not supported yet: ../../../../R-3.5.1/src/main/array.c:2178

Data is not bit (215 rows per 45 variables) and there are 34 random intercepts in the model. Don't know if it is related to a R upgrade or not, but it started appearing just recently.

The glmer call is:
`glmer(cbind(TOTOK, 12 - TOTOK) ~ DefHTA + (1 | Scuola), family = binomial(), data = Data) %>% confint.default()`

The attached data is produced with a dump() call.
[put.txt](https://github.com/lme4/lme4/files/2238203/put.txt)

Thanks

**Comment by bbolker** [comment]
**tl;dr**:`confint.default()` just won't work on `merMod` objects (see below), but you can get Wald confidence intervals (which are fast and approximate and are what `confint.default()` would give you anyway) via  `confint(g1,method="Wald")`.

---
I get a different error:

> Error in diag(vov(object)) :  no method for coercing this S4 class to a vector

On the surface this is a little weird, since `diag(vcov(g1))` (I assigned the `glmer` fit to the `g1` variable) works when run in the global environment. I don't know why it doesn't work in the context of `confint.default()` (it fails if I run it interactively within the debugger while running `confint.default`).

The proximal problem is that `vcov()` returns a `dpoMatrix` object rather than a simple base-R matrix. I've thought for some time that we should consider casting to a regular matrix before returning from `vcov.merMod()` - I don't see much advantage to `dpoMatrix` here, and it does cause various minor compatibility problems (@mmaechler , your opinions?)

However, even if we understood/fixed this problem, `confint.default()` still wouldn't work for `merMod` objects, since it assumes that the `coef()` method returns a vector of fixed-effect parameters, whereas `lme4` (and other mixed model packages following the conventions established by `nlme`) use `coef()` to return a list of group-level parameter estimates, while `fixef()` is used for the population-level fixed-effect parameters. (This is why `confint.merMod` has a "Wald" option ...)

**Comment by bakaburg1** [comment]
Dear Ben, thanks for the help.

This is sessionInfo() after a R restart():

```
R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] it_IT.UTF-8/it_IT.UTF-8/it_IT.UTF-8/C/it_IT.UTF-8/it_IT.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.5.1 tools_3.5.1    yaml_2.1.19   

```

And this after library() lme4 and Matrix (I reported only the changed parts):

```
other attached packages:
[1] lme4_1.1-17   Matrix_1.2-14

loaded via a namespace (and not attached):
 [1] minqa_1.2.4     MASS_7.3-50     compiler_3.5.1  tools_3.5.1     yaml_2.1.19     Rcpp_0.12.17    splines_3.5.1   nlme_3.1-137    grid_3.5.1     
[10] nloptr_1.0.4    lattice_0.20-35
```
And finally in my active session, just in case there is a interaction between package problem (sorry, tidyverse stuff here):

```
attached base packages:
[1] grid      parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2   partykit_1.2-2   mvtnorm_1.0-8    libcoin_1.0-1    rpart.plot_3.0.2 rpart_4.1-13     blme_1.0-4       plyr_1.8.4       arm_1.10-1      
[10] lme4_1.1-17      Matrix_1.2-14    MASS_7.3-50      rmarkdown_1.10   knitr_1.20       readxl_1.1.0     lubridate_1.7.4  scales_0.5.0     broom_0.5.0     
[19] magrittr_1.5     forcats_0.3.0    stringr_1.3.1    dplyr_0.7.6      purrr_0.2.5      readr_1.1.1      tidyr_0.8.1      tibble_1.4.2     ggplot2_3.0.0   
[28] tidyverse_1.2.1 

loaded via a namespace (and not attached):
 [1] minqa_1.2.4         colorspace_1.3-2    class_7.3-14        rprojroot_1.3-2     htmlTable_1.12      pls_2.6-0           base64enc_0.1-3    
 [8] rstudioapi_0.7      DRR_0.0.3           SnowballC_0.5.1     prodlim_2018.04.18  xml2_1.2.0          codetools_0.2-15    splines_3.5.1      
[15] robustbase_0.93-1   RcppRoll_0.3.0      Formula_1.2-3       jsonlite_1.5        nloptr_1.0.4        caret_6.0-80        ddalpha_1.3.4      
[22] cluster_2.0.7-1     kernlab_0.9-26      sfsmisc_1.1-2       compiler_3.5.1      httr_1.3.1          backports_1.1.2     assertthat_0.2.0   
[29] lazyeval_0.2.1      cli_1.0.0           acepack_1.4.1       htmltools_0.3.6     tools_3.5.1         coda_0.19-1         gtable_0.2.0       
[36] glue_1.2.0          reshape2_1.4.3      Rcpp_0.12.17        cellranger_1.1.0    nlme_3.1-137        iterators_1.0.9     timeDate_3043.102  
[43] inum_1.0-0          gower_0.1.2         rvest_0.3.2         DEoptimR_1.0-8      ipred_0.9-6         hms_0.4.2           RColorBrewer_1.1-2 
[50] yaml_2.1.19         gridExtra_2.3       latticeExtra_0.6-28 stringi_1.2.3       highr_0.7           foreach_1.4.4       checkmate_1.8.5    
[57] lava_1.6.2          geometry_0.3-6      rlang_0.2.1         pkgconfig_2.0.1     evaluate_0.10.1     lattice_0.20-35     bindr_0.1.1        
[64] htmlwidgets_1.2     recipes_0.1.3       CVST_0.2-2          tidyselect_0.2.4    R6_2.2.2            Hmisc_4.1-1         dimRed_0.1.0       
[71] pillar_1.2.3        haven_1.1.2         foreign_0.8-70      withr_2.1.2         survival_2.42-3     abind_1.4-5         nnet_7.3-12        
[78] modelr_0.1.2        crayon_1.3.4        utf8_1.1.4          data.table_1.11.4   ModelMetrics_1.1.0  digest_0.6.15       stats4_3.5.1       
[85] munsell_0.5.0       magic_1.5-8        
```

**Comment by bakaburg1** [comment]
But here's something that worries me even more!

```r
> glmer(cbind(TOTOK, 12 - TOTOK) ~ DefHTA + (1 | Scuola), family = binomial(), data = Data) %>% confint(method = 'Wald')
                                        2.5 %     97.5 %
.sig01                                     NA         NA
(Intercept)                         0.7272197  0.9232485
DefHTAHighly Tecnical Assessment   -2.2385820 -0.5478395
DefHTAHealth Threat Administration -1.6067737  0.7256851
DefHTANon so                       -2.0643328 -1.1029170
> glmer(cbind(TOTOK, 12 - TOTOK) ~ DefHTA + (1 | Scuola), family = binomial(), data = Data) %>% confint
Computing profile confidence intervals ...
                                        2.5 %     97.5 %
.sig01                              0.0000000  0.2436033
(Intercept)                         0.7243622  0.9257465
DefHTAHighly Tecnical Assessment   -2.2788021 -0.5656011
DefHTAHealth Threat Administration -1.6000654  0.7937890
DefHTANon so                       -2.0781847 -1.1126456
> glmer(cbind(TOTOK, 12 - TOTOK) ~ DefHTA + (1 | Scuola), family = binomial(), data = Data) %>% tidy(conf.int = T)
# A tibble: 5 x 8
  term                               estimate std.error statistic   p.value conf.low conf.high group 
  <chr>                                 <dbl>     <dbl>     <dbl>     <dbl>    <dbl>     <dbl> <chr> 
1 (Intercept)                           0.825    0.0500    16.5    3.55e-61    0.727     0.923 fixed 
2 DefHTAHighly Tecnical Assessment     -1.39     0.431     -3.23   1.24e-3   -2.24     -0.548 fixed 
3 DefHTAHealth Threat Administration   -0.441    0.595     -0.740  4.59e-1   -1.61      0.726 fixed 
4 DefHTANon so                         -1.58     0.245     -6.46   1.07e-10   -2.06     -1.10  fixed 
5 sd_(Intercept).Scuola                 0.118   NA         NA     NA          NA        NA     Scuola
```

broom::tidy() gives the intervals with the Wald method with merMod objects! while with glm use profile intervals like confint()!!! Until now I always the CIs I got from tidy were profile ones!

**Comment by bbolker** [comment]
Nothing jumps out at me about your `sessionInfo()`.  Do you see the same symptoms even with the clean session?

As for `broom` ... I'm in the process of putting together `broom.mixed`, see `https://github.com/bbolker/broom.mixed`.  The argument for making Wald the default tidying confidence interval is that computing the profile confidence intervals can be very slow for large models ...

**Comment by bbolker** [comment]
Since I can't reproduce this, and since the original problems seem to be dealt with, I'm going to close it - will happily reopen if necessary.
